### ISSUE TITLE
Make it possible to layer a facet chart

### ISSUE THREAD

**Issue by mattijn** [altair-viz/altair#2878]
Suggestion for enhancement based on the discussion that started here: https://github.com/altair-viz/altair/pull/2874:

The following altair specification will fail:
```python
import altair as alt
from vega_datasets import data
stocks = data.stocks.url

c1 = alt.Chart(stocks, height=40, width=80).encode(
    x='date:T',
    y='price:Q',
    column='symbol:N'
)

c1.mark_line() + c1.mark_point()
```

But this works:
```python
c1.mark_line(points=True)
```

The last one works, because Vega-Lite is actually¹ translating this into:
```python
c2 = alt.Chart(stocks, height=40, width=80).encode(
    x='date:T',
    y='price:Q'
)

(c2mod.mark_line() + c2.mark_point()).facet(column='symbol:N')
```

Is it feasible or desirable to do the same lifting on the Altair side?

> ¹ 
> In the Vega-Lite specification it is just `mark_line(point=True)`, but if you observe the Extended Vega-Lite spec in the Vega-Editor it does creates a layered chart and shifts the `column` encoding channel to a top-level `facet` operator, see [Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4Igxg9gdgZglgcxALlANzgUwO4tJKAFzigFcJSBnAdTgBNCALFAZgAY2AacaYsiygAlMiRoVYcAvpO50AhoTl4QpAE4AbFCDGEADpWQB6Q2DpQAdACtKdTOrhpV5qJkKGougLaG0mBHIBaeUVKV0oAATQARnMAJgBOczZDYLlDSkIIMABrSnMwSjQQGRBPOVVs5UIAT11MLXsXEG5dCBJxZEJVUkwSzChIOhIkVB51Uk8oZXg7Oi1Kas8AIwhNbhq6rSgITxI5TRKAD2msdTnkEGD69dr6i8JMT1bVfeLuapPZrV1VODBrkAbO4gACOpDkRDgimIvmKJUYIgQYhQABYuCBsPQmCgABzogAklDACLKWh0+iMPj8cnMCChjFIS3McAg6WJjzSvn8AXsDx8AFY4kkrJRoMUgA), click bottom-left next to Compiled Vega.
>
> _Originally posted by @mattijn in https://github.com/altair-viz/altair/issues/2874#issuecomment-1421462825_:

**Comment by ChristopherDavisUCI** [comment]
Hi @mattijn, my first impression is to be pretty hesitant to add this kind of extra feature to Altair, just because these extras seem to cause the most difficulties in regards to maintenance (like allowing parameters to be added in more places than Vega-Lite does).  Do you see any reason this would be easier to accomplish in Altair vs in Vega-Lite?  If not, I'd vote for trying to add it to Vega-Lite instead if it's a feature we'd like.

I believe this is where the translation you mentioned for `point=True` happens in Vega-Lite: https://github.com/vega/vega-lite/blob/next/src/normalize/pathoverlay.ts

**Comment by mattijn** [comment]
Even better if we can solve it on the vega-lite side.

I still have to get used to a short vega-lite specification and the possibility of expanding or extending a vega-lite specification.

In hindsight maybe we could have exploited that more for the parameters as well.

**Comment by ChristopherDavisUCI** [comment]
I agree!  Would you like to open an issue in Vega-Lite, partially to hear if this is something they see as plausible?

**Comment by joelostblom** [comment]
My vote would also be to try to solve things like this that are within the scope of Vega-Lite on their side.

**Comment by ChristopherDavisUCI** [comment]
I'll repost in this message if I find a more fleshed out explanation, but here is a brief one by Jake: https://stackoverflow.com/a/64422039

> A layer chart is not allowed to contain a facet encoding... (the reason for this is that the semantics of layers containing incompatible facets is unclear).

**Comment by kanitw** [comment]
+1 to the answer above.  (I'm a co-author of Vega-Lite.)

**Comment by mattijn** [comment]
Thanks @ChristopherDavisUCI for bringing info why this is currently not supported. Thanks @kanitw for chiming in!

I hope you don't mind if I ask one more question. 
I understand this will not work when there are different incompatible facets defined within the subcharts (that should raise an error) but in the case of above, the facets are identical and compatible. Will this trigger the same semantic issue?

**Comment by dangotbanned** [comment]
@mattijn @kanitw @ChristopherDavisUCI @joelostblom 

I'm having a tough time figuring out if this issue and one in `vega-lite` are related

- https://github.com/vega/vega-lite/issues/4677

Is this the same or the inverse?

**Comment by joelostblom** [comment]
I think that issue is the inverse, in general it works well to facet layered charts but not to layer feacetted charts.
